### DISCUSSION TITLE
Push notifications to deep urls not working if app not open

### DISCUSSION THREAD

**Discussion by andreas-globi** [buildnatively/community#71] [Ideas]
### Please confirm you've checked the following resources:

- [X] I've searched through the [Docs](https://docs.buildnatively.com).
- [X] I've checked [Natively Youtube Playlist 1](https://youtube.com/playlist?list=PLYo9EJqlVMNdtzb_rm6_6kjlDCmaoAqlO&si=sOygZtRDGUn_2Yi8) and [Natively Youtube Playlist 2](https://youtube.com/playlist?list=PLYo9EJqlVMNcwz2o1Xx5xVYYzzKgcyMeB&si=f4GuCPvUAdjvHplD).
- [X] I've reviewed the [Troubleshooting Guides](https://docs.buildnatively.com/guides/troubleshooting).
- [X] I've visited the [FAQ](https://docs.buildnatively.com/natively-platform/faq).

### Which web app platform are you using?

Other

### Which SDK are you using?

JavaScript SDK

### JS SDK / Bubble Plugin Version (e.g., 2.12.1)

2.11.0

### Platform with the issue?

Android

### App ID

1704564868698x757430365325099000

### Steps to reproduce

1. Send push notifications to a deep page, eg: https://myapp.com/page/id/ using the url parameter with one signal to a specific player ID
2. click the notification on the device

### Expected results

The app should open to the page of the URL provided in the notification

### Actual results

If the app is already open on the device, it opens to the correct page

If the app is not currently open, it opens to the configured homepage

**Comment by romanfurman6** [comment]
@andreas-globi Hey, can you please provide more details on this issue. 
1. What is your Android version? 
2. What is a version of Natively App (Can be found in 'Natively Info' SDK version param)

**Comment by andreas-globi** [comment]
I'm on Android 14 (Galaxy S22)

I'm not using the Natively app - I'm using the built and installed APK.

**Reply by romanfurman6** [reply]
okay, let me check thanks!

**Reply by romanfurman6** [reply]
Hello @andreas-globi, 

I have checked the latest version of Natively on Android 14 (Galaxy S21) and it appears to be working fine for me in both cases. Could you please provide more information so that I can understand what might be causing the issue? Can you also check if the URL you're sending has the same domain? 

Thank you.

**Reply by andreas-globi** [reply]
I have verified that the URL being sent is definitely correct.

Here are 3 methods of testing and the results:

1) Mobile app is open. Hit the Home key to get back to the launcher screen. Select a notification. End up on the correct page 100% of the time

2) In the mobile app, hit the Back button until it closes and I'm back at the launcher screen. Select a notification. End up at the app's homepage 80-90% of the time. Occasionally though I do end up on the correct page.

3) Force close mobile app. Select a notification. End up at the correct page 80% of the time. But about 20% of the time I end up on the app's homepage.

These results are inconsistent and I don't know why.

In my testing I send 2 notifications - always the same 2 and to 2 different urls (so that caching doesn't come into play).

By far the most error-prone is hitting back until the app soft closes.

**Reply by romanfurman6** [reply]
@andreas-globi thanks for sharing this, let me check and get back to you

**Reply by andreas-globi** [reply]
Any updates on this?

**Reply by romanfurman6** [reply]
Hello, @andreas-globi ! Will let you know today 💯

**Reply by romanfurman6** [reply]
i see the issue, not sure if we will be able to fix it in this version or need to wait till Natively 3.0 (+-2mo)
I will get back to you with a feedback

**Reply by romanfurman6** [reply]
@andreas-globi  Hey, it seems fixed now. Can you please rebuild the app and let us know if that helped 🙏

**Reply by andreas-globi** [reply]
Just tested and it seems to work as expected.

Thanks !
